pandera/backends/pyspark/container.py

```diff
@@ -490,6 +490,82 @@ def _try_coercion(obj, colname, col_schema):
 
         return obj
 
+    @validate_scope(scope=ValidationScope.DATA)
+    def unique(
+        self,
+        check_obj: DataFrame,
+        *,
+        schema=None,
+        error_handler: ErrorHandler = None,
+    ):
+        """Check uniqueness in the check object."""
+        assert schema is not None, "The `schema` argument must be provided."
+        assert (
+            error_handler is not None
+        ), "The `error_handler` argument must be provided."
+
+        if not schema.unique:
+            return check_obj
+
+        try:
+            check_obj = self._check_uniqueness(
+                check_obj,
+                schema,
+            )
+        except SchemaError as err:
+            if not error_handler.lazy:
+                raise err
+            error_handler.collect_error(
+                ErrorCategory.DATA, err.reason_code, err
+            )
+
+        return check_obj
+
+    def _check_uniqueness(
+        self,
+        obj: DataFrame,
+        schema,
+    ) -> DataFrame:
+        """Ensure uniqueness in dataframe columns.
+
+        :param obj: dataframe to check.
+        :param schema: schema object.
+        :returns: dataframe checked.
+        """
+        # Use unique definition of columns as first option
+        # unique_columns = [col.unique for col in schema.columns.values()]
+
+        # Overwrite it, if schemas's Config class has a unique declaration
+        unique_columns = (
+            [schema.unique]
+            if isinstance(schema.unique, str)
+            else schema.unique
+        )
+
+        duplicates_count = (
+            obj.select(*unique_columns)  # ignore other cols
+            .groupby(*unique_columns)
+            .agg(count("*").alias("pandera_duplicate_counts"))
+            .filter(
+                col("pandera_duplicate_counts") > 1
+            )  # long name to avoid colisions
+            .count()
+        )
+
+        if duplicates_count > 0:
+            raise SchemaError(
+                schema=schema,
+                data=obj,
+                message=(
+                    f"Duplicated rows [{duplicates_count}] were found "
+                    f"for columns {unique_columns}"
+                ),
+                check="unique",
+                reason_code=SchemaErrorReason.DUPLICATES,
+            )
```

Comment:
```suggestion
        # Determine unique columns based on schema's config
        unique_columns = [schema.unique] if isinstance(schema.unique, str) else schema.unique
        
        duplicates_count = (
            obj.select(*unique_columns)
            .groupby(*unique_columns)
            .agg(count("*").alias("pandera_duplicate_counts"))
            .filter(col("pandera_duplicate_counts") > 1)
            .count()
        )

        if duplicates_count > 0:
            raise SchemaError(
                schema=schema,
                data=obj,
                message=(
                    f"Duplicated rows found ({duplicates_count}) in columns {unique_columns}"
                ),
                check="unique",
                reason_code=SchemaErrorReason.DUPLICATES,
            )

        return obj
```suggestion
        unique_columns = (
            [schema.unique]
            if isinstance(schema.unique, str)
            else schema.unique
        )

        duplicates_count = (
            obj.select(*unique_columns)  # ignore other cols
            .groupby(*unique_columns)
            .agg(count("*").alias("pandera_duplicate_counts"))
            .filter(
                col("pandera_duplicate_counts") > 1
            )  # long name to avoid colisions
            .count()
        )

        if duplicates_count > 0:
            raise SchemaError(
                schema=schema,
                data=obj,
                message=(
                    f"Duplicated rows [{duplicates_count}] were found "
                    f"for columns {unique_columns}"
                ),
                check="unique",
                reason_code=SchemaErrorReason.DUPLICATES,
            )
```
```suggestion
        unique_columns = (
            [schema.unique]
            if isinstance(schema.unique, str)
            else schema.unique
        )

        duplicates_count = (
            obj.select(*unique_columns)  # ignore other cols
            .groupby(*unique_columns)
            .agg(count("*").alias("pandera_duplicate_counts"))
            .filter(
                col("pandera_duplicate_counts") > 1
            )  # long name to avoid colisions
            .count()
        )

        if duplicates_count > 0:
            raise SchemaError(
                schema=schema,
                data=obj,
                message=(
                    f"Duplicated rows [{duplicates_count}] were found "
                    f"for columns {unique_columns}"
                ),
                check="unique",
                reason_code=SchemaErrorReason.DUPLICATES,
            )
```
```suggestion
        unique_columns = (
            [schema.unique]
            if isinstance(schema.unique, str)
            else schema.unique
        )

        duplicates_count = (
            obj.select(*unique_columns)  # ignore other cols
            .groupby(*unique_columns)
            .agg(count("*").alias("pandera_duplicate_counts"))
            .filter(
                col("pandera_duplicate_counts") > 1
            )  # long name to avoid colisions
            .count()
        )

        if duplicates_count > 0:
            raise SchemaError(
                schema=schema,
                data=obj,
                message=(
                    f"Duplicated rows [{duplicates_count}] were found "
                    f"for columns {unique_columns}"
                ),
                check="unique",
                reason_code=SchemaErrorReason.DUPLICATES,
            )
```
```

Reply:
I think it would make sense to cleanup the first row a bit

Reply:
Clean up done. Let me know if you have any other suggestion in this part.